### PR TITLE
fix(core): don't strip shell/YAML # unless it starts a comment

### PR DESCRIPTION
**Pull request by serhiizghama**
`--remove-comments` silently corrupts shell and YAML files. Both are routed to the `perl` strip profile, which treats every unquoted `#` as a line comment. But in shell and YAML a `#` is only a comment at the start of a line or after whitespace — not inside `${name##*/}`, `$#`, or an unquoted URL like `a/b#readme`.

So a shell script like:

```sh
base=${name##*/}
ext=${name#*.}
count=$#
```

comes out as `base=${name`, `ext=${name`, `count=$` — the rest of each line is dropped at the first `#`. Same for YAML: `repo: https://github.com/a/b#readme` loses `#readme`. The packed output no longer matches the source, which is exactly what you don't want when feeding it to an LLM.

I replaced the `perl` routing for `.sh`, `.yaml`, and `.yml` with a small boundary-aware scanner that only cuts at a `#` when it's at line start or preceded by whitespace, and skips single/double quoted strings (so `echo "a # b"` and `color: "#ff0000"` are kept). Real comments — full-line, whitespace-preceded trailing ones, and shebangs — are still stripped, so the existing shell/YAML tests stay green. Added tests for the parameter-expansion, `$#`, quoted-hash, and URL-fragment cases.

### CODE REVIEW

**Comment by serhiizghama** [comment]
Good catch — heredocs and block scalars are literal content, the boundary rule shouldn't touch them. Added line/indentation-scoped tracking for both (shell `<<EOF`...`EOF` bodies, YAML `key: |` bodies) so they're copied verbatim instead of scanned for `#`. Added regression tests for both, verified they fail on the unfixed scanner.

**Comment by serhiizghama** [comment]
@coderabbitai review

**Comment by serhiizghama** [comment]
Good catches on the YAML edge cases. Fixed both: quotes now only open a scalar at a value boundary in YAML, so an apostrophe in a plain scalar (`note: it's fine  # x`) no longer flips the quote state and eats the following comments; and the block-scalar regex now requires the `|`/`>` to follow a `:`/`-` introducer so a plain scalar that merely ends in `|` isn't treated as a block opener. Added regression tests for both.

Left `rtrimLines` as-is — it's applied to every language's stripped output, not just shell/YAML, so trailing whitespace in packed content is already normalized project-wide; carving out heredoc/block-scalar bodies would make the behavior inconsistent for a tool whose output is meant for reading, not byte-exact reconstruction.

**Comment by serhiizghama** [comment]
@coderabbitai review

File: src/core/file/fileManipulate.ts [hunk 1]
```diff
@@ -42,6 +42,131 @@ class StripCommentsManipulator extends BaseManipulator {
...
+          heredocStripLeadingTabs = match[1] === '-';
+          heredocDelimiter = match[3];
+        }
+      } else if (this.language === 'yaml' && !inSingle && !inDouble && YAML_BLOCK_SCALAR_START.test(processed.text)) {
+        blockScalarIndent = line.match(/^[ \t]*/)?.[0].length ?? 0;
+      }
+    }
+
+    return rtrimLines(output.join('\n'));
+  }
```
[the remark this answers is not in the document]
Comment (serhiizghama) [reply src/core/file/fileManipulate.ts:112]: rtrimLines runs over the whole output as a normalization step, and StripCommentsManipulator does the same on line 41 — so trailing whitespace is trimmed from every language repomix processes, not just here. What this PR preserves verbatim is the `#` inside heredoc/block-scalar bodies (which would otherwise be stripped as a comment); trailing-whitespace normalization is a separate tool-wide behavior. Keeping heredoc bodies with trailing whitespace while every other file type gets trimmed would be the inconsistency, so I'd rather leave rtrimLines as-is here.

**Review by yamadashy — APPROVED** [review]

**Comment by yamadashy** [comment]
@serhiizghama Thanks, the boundary rule and the heredoc/block scalar handling are solid. I pushed a few more edge-case tests on top. Merging 🙏
